crates/dbsp/src/algebra/decimal.rs

```diff
@@ -0,0 +1,607 @@
+use num::{FromPrimitive, Num, One, Signed, ToPrimitive, Zero};
+use rust_decimal::{Decimal, MathematicalOps};
+use size_of::SizeOf;
+use std::hash::Hash;
+use std::{
+    fmt::Display,
+    ops::{Add, AddAssign, Div, Mul, Neg, Rem, Sub},
+    str::FromStr,
+};
+
+mod inner_types {
+    use rust_decimal::Decimal;
+    use size_of::SizeOf;
+    use std::fmt::Display;
+
+    const MIN_DECIMAL_PRECISION: u32 = 1;
+    const MAX_DECIMAL_PRECISION: u32 = 38;
+
+    const MIN_DECIMAL_SCALE: u32 = 0;
+    const MAX_DECIMAL_SCALE: u32 = 28;
+
+    #[derive(
+        Debug,
+        Clone,
+        Copy,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Hash,
+        SizeOf,
+        rkyv::Archive,
+        rkyv::Serialize,
+        rkyv::Deserialize,
+    )]
+    #[cfg_attr(feature = "with-serde", derive(serde::Serialize, serde::Deserialize))]
+    pub struct Precision(u32);
+
+    impl Precision {
+        pub fn new(val: Decimal) -> Result<Self, String> {
```

Comment:
This shouldn't be called "new", but rather "extract_precision_from_decimal".
What is the meaning of "precision" in this case? What if the decimal is something like 5.0000?
I would expect that you never need this method, the compiler should always tell you what the precision of something is.

Reply:
Yup, don't need it now. Removing it.